packages/server/lib/controllers/v1/account/validateEmailAndLogin.ts

```diff
@@ -0,0 +1,74 @@
+import { z } from 'zod';
+import { asyncWrapper } from '../../../utils/asyncWrapper.js';
+import { getLogger, requireEmptyQuery, zodErrorToHTTP } from '@nangohq/utils';
+import { analytics, userService, AnalyticsTypes, environmentService, createOnboardingProvider } from '@nangohq/shared';
+import type { WebUser, ValidateEmailAndLogin } from '@nangohq/types';
+
+const logger = getLogger('Server.ValidateEmailAndLogin');
+
+const validation = z
+    .object({
+        token: z.string().min(6)
+    })
+    .strict();
+
+export const validateEmailAndLogin = asyncWrapper<ValidateEmailAndLogin>(async (req, res) => {
+    const emptyQuery = requireEmptyQuery(req);
+
+    if (emptyQuery) {
+        res.status(400).send({ error: { code: 'invalid_query_params', errors: zodErrorToHTTP(emptyQuery.error) } });
+        return;
+    }
+
+    const val = validation.safeParse(req.body);
+
+    if (!val.success) {
+        res.status(400).send({
+            error: { code: 'invalid_body', errors: zodErrorToHTTP(val.error) }
+        });
+        return;
+    }
+
+    const { token } = val.data;
+
+    const userAndAccount = await userService.getUserAndAccountByToken(token);
+
+    if (!userAndAccount) {
+        logger.error('Error validating user');
+        res.status(500).send({ error: { code: 'error_validating_user', message: 'There was a problem validating the user. Please reach out to support.' } });
+        return;
+    }
+
+    await userService.verifyUserEmail(userAndAccount.user_id);
```

Comment:
what happens if this fails?

Reply:
This is just an update in the database, in the unlikely event it fails it'll hit the catch as provided by the `asyncWrapper`